controllers/fenceagentsremediation_controller.go

```diff
@@ -325,6 +337,47 @@ func (r *FenceAgentsRemediationReconciler) updateStatus(ctx context.Context, far
 	return nil
 }
 
+func (r *FenceAgentsRemediationReconciler) collectSecretParams(far *v1alpha1.FenceAgentsRemediation, ctx context.Context) (map[string]string, error) {
+	secretParams := map[string]string{}
+
+	sharedSecretName := far.Spec.SharedSecretName
+	if len(sharedSecretName) > 0 {
+		sharedSecret, err := r.getSecret(ctx, client.ObjectKey{Name: sharedSecretName, Namespace: far.Namespace})
+		if err != nil && !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "failed to fetch secret", "secret name", sharedSecretName, "namespace", far.Namespace)
+			return nil, fmt.Errorf(errorFailFetchingSecret, sharedSecretName, far.Namespace, err)
+		}
+
+		//Fill secret params from shared secret
+		if sharedSecret != nil {
+			for secretKey, secretVal := range sharedSecret.Data {
+				secretParams[secretKey] = string(secretVal)
```

Comment:
Does it include any [decryption](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#ensure-all-secrets-are-encrypted)? @mshitrit 
IIUC, we store the plaintext values from the secret in secretParams.

Reply:
k8s automatically decrypts Secrets at the API server level so it's not something we need to manage, and since we don't create the Secrets IIUC it's up to the user to decide whether to encrypt the data or not.